proofs/proof_type.mli

```diff
@@ -26,3 +26,8 @@ type rule = prim_rule
 type goal = Goal.goal
 
 type tactic = goal sigma -> goal list sigma
+(** @deprecated: Please use "Tacmach.tactic" type instead.
+
+    We keep this definition only for "technical reasons".
+    If one tries to get rid of it, Ocaml compiler complains
+    about "circular dependency" between "Refiner" and "Tacmach" modules. *)
```

Comment:
@ppedrot Is this the correct thing to say?

Previously, I tried to expose this type in the API and you subsequently corrected that mistake. A comment of this sort would help me to remember that this is a bad idea.